backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/handler/ServiceHandler.kt

```diff
@@ -61,14 +79,68 @@ class ServiceHandler @Autowired constructor(
 
     }
 
-    fun findAll(pageNumber: Int, pageSize: Int): ResultPage<Service> =
-            servicePersistor.findAll(pageNumber, pageSize)
+    fun findAll(pageNumber: Int, pageSize: Int): ResultPage<Service> {
+        val userId = jwtAuthenticationProvider.getCurrentUserId()
+        val page = servicePersistor.findAll(pageNumber, pageSize, userId, apiCenterProperties.getAnonymousUsername())
+        return ResultPage(this.filterByViewPermission(page.content), page.last)
```

Comment:
I could imagine either something like a "findAllWithPermissions" method or a `filter` method on `Page`.

Reply:
This pre-release filter function is used in other places in the handler, for non-paged requests as well, so I would like to keep it in this class. And I didn't want to put this logic in the servicePersistor, because it is permissions logic, and not strictly the responsibility of a storage service. (Of course, I had to compromise on that a bit, because the paging request query needs to know which user is logged in, so that the number of services per page is consistent regardless of who is logged in... 🤷‍♂)